src/goto-synthesizer/cegis_verifier.cpp

```diff
@@ -53,6 +53,37 @@ static bool contains_symbol_prefix(const exprt &expr, const std::string &prefix)
   return false;
 }
 
+static const exprt &
+get_checked_pointer_from_null_pointer_check(const exprt &violation)
+{
+  // A NULL-pointer check is the negation of an equation between the checked
+  // pointer and a NULL pointer.
+  // ! (POINTER_OBJECT(NULL) == POINTER_OBJECT(ptr))
+  const equal_exprt &equal_expr = to_equal_expr(to_not_expr(violation).op());
+
+  const exprt &lhs_pointer_object = equal_expr.lhs();
+  const exprt &rhs_pointer_object = equal_expr.rhs();
+  INVARIANT(
+    lhs_pointer_object.id() == ID_pointer_object &&
+      rhs_pointer_object.id() == ID_pointer_object,
+    "Null pointer check is an equation between two pointer objects of "
+    "pointer-type operands.");
+
+  const exprt &lhs_pointer = lhs_pointer_object.operands()[0];
+  const exprt &rhs_pointer = rhs_pointer_object.operands()[0];
```

Comment:
In the same vein as above: use `to_pointer_object_expr` to get the invariant check for free.